spring-boot/src/main/java/org/springframework/boot/jta/narayana/NarayanaRecoveryManagerBean.java

```diff
@@ -41,8 +40,8 @@ public NarayanaRecoveryManagerBean(RecoveryManagerService recoveryManagerService
 		this.recoveryManagerService = recoveryManagerService;
 	}
 
-	@Override
-	public void afterPropertiesSet() throws Exception {
+	@EventListener
+	public void create(ApplicationReadyEvent ignored) throws Exception {
```

Comment:
Moving this to application ready makes me a little nervous. What is the implication for any transactional logic that tries to run earlier? That could even be in another listener to the `ApplicationReadyEvent` that happens to run earlier than this one.